pypesto/startpoint/base.py

```diff
@@ -18,31 +19,189 @@ class StartpointMethod:
     def __call__(
         self,
         n_starts: int,
+        problem: Problem,
+    ) -> np.ndarray:
+        """Generate startpoints.
+
+        Parameters
+        ----------
+        n_starts: Number of starts.
+        problem: Problem specifying e.g. dimensions, bounds, and guesses.
+
+        Returns
+        -------
+        xs: Startpoints, shape (n_starts, n_par).
+        """
```

Comment:
`NotImplementedError()`?

Reply:
To raise that, you'd have to make the method non-abstract. As this is an interface class never expected to provide an implementation, I'd prefer abstract.

Reply:
https://docs.python.org/3/library/exceptions.html#NotImplementedError ? I thought thats the standard python way of marking the method abstract.

Reply:
afaik: Conceptually, `NotImplementedError` indicates that something has not (yet) been implemented, which can have various reasons, including lazyness of the developer, or abstracness. If the whole class however is to be considered abstract, `@abstractmethod` is preferable, which prevents any instantiation of the class in the first place until all abstract methods have been implemented (in derived classes).